File: src/Transction.cpp
#include <iostream>
#include <string>
#include "./User.cpp"
#include "./Book.cpp"
#include "./Librarian.cpp"

using namespace std;

class Transaction
{
private:
    Book *borrowedBook;
    Book *returnedBook;
    double fine;
    Book *book;
    User *user;
    Librarian *librarian;

public:
    Transaction(Book *transactionBook, User *transactionUser, Librarian *transactionLibrarian);

    Transaction::Transaction(
        Book *transactionBook,
        User *transactionUser,
        Librarian *transactionLibrarian)
        : //   borrowedDate(time(0)),
          //   returnedDate(0),
          fine(0.0),
          book(transactionBook),
          user(transactionUser),
          librarian(transactionLibrarian)
    {
    }
};

File: src/Transction.js
import User from "./User.js";
import Book from "./Book.js";
import Librarian from "./Librarian.js";

class Transaction {
    constructor(transactionBook, transactionUser, transactionLibrarian) {
        this.borrowedBook = null; 
        this.returnedBook = null; 
        this.fine = 0.0;
        this.book = transactionBook;
        this.user = transactionUser;
        this.librarian = transactionLibrarian;
    }
}

export default Transaction;
